### ISSUE TITLE
IDISP013 should not warn on using calling ctor with func<task>

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#71]
```cs
namespace RoslynSandbox
{
    using System;
    using System.Threading.Tasks;

    public struct Bar : IDisposable
    {
        public Bar(Func<Task> task)
        {
            this.Task = task();
        }

        public Task Task { get; }

        public void Dispose()
        {
        }
    }

    public sealed class Foo
    {
        public Foo()
        {
            var fromResult = Task.FromResult(1);
            using (var bar = new Bar(() => fromResult))
            {
            }
        }
    }
}
```
